File: examples/nodejs/silent_cron_failures.js
#!/usr/bin/env node
/**
 * Silent Cron Failures
 * Problem: Cron job fails silently without producing error logs
 * Solution: Explicit success confirmation with ping
 * Documentation: https://deadmanping.com/silent-cron-failures
 */

const https = require('https');

const MONITOR_ID = 'YOUR_MONITOR_ID';  // Replace with your actual monitor ID

// Your actual work
async function run() {
  await performBackup();
  await syncData();
  
  // Single ping at end - if job fails, ping won't arrive and DeadManPing will alert
  https.request(`https://deadmanping.com/api/ping/${MONITOR_ID}`, { method: 'POST' }).end();
}

run().catch((err) => {
  // If job fails, ping won't arrive - DeadManPing will detect missing ping
  process.exit(1);
});

File: examples/python/silent_cron_failures.py
#!/usr/bin/env python3
"""
Silent Cron Failures
Problem: Cron job fails silently without producing error logs
Solution: Explicit success confirmation with ping
Documentation: https://deadmanping.com/silent-cron-failures
"""

import requests
import sys

MONITOR_ID = "YOUR_MONITOR_ID"  # Replace with your actual monitor ID

# Your actual work
perform_backup()
sync_data()

# Single ping at end - if job fails, ping won't arrive and DeadManPing will alert
requests.post(f"https://deadmanping.com/api/ping/{MONITOR_ID}")
